docs/design/crd-validation/README.md

```diff
@@ -0,0 +1,79 @@
+## Background
+
+This proposal is to add support for admission controllers in OCS. An admission controller is a piece of code that intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized
+
+ There are two special controllers: MutatingAdmissionWebhook and ValidatingAdmissionWebhook.
+ 
+ Mutating controllers may modify the objects they admit but validation controllers are only allowed to validate requests.
+
+## User Scenario
+
+Listing out some use cases, where validators will be helpful,
+
+**Securing the resources from CLI updates**
+
+Currently, user can manipulate Custom Resource specs with any values and apply them through CLI which may result in OCS not functioning as expected. The present validation method in Kubernetes is the OpenAPI schema validation which can be used for basic validations like checking type of data, providing a range for the values etc but anything more complex (checking resource availability, network status, error handling) would not be possible under this scenario.
+
+**Expansion/Adding Capacity**
+Making sure that while adding capacity of an OCS cluster, the storage class that will be selected should have the same provisioner value as of the SC that was used to create the OCS cluster.
+
+## Admission Controllers Concept
+
+![AC Image](ac_design.png)
+
+## Requirements
+
+1. Webhook server which will validate the requests
+2. TLS certificates for the server,
+3. ValidatingWebhookConfig/MutatingWebhookConfig which will intercept requests and send a HTTPS request to the webhook server.
+4. RBAC Components 
+
+## Implementation 
+
+As shown in the above diagram, the admission control process proceeds in two phases. In the first phase, mutating admission controllers are run. In the second phase, validating admission controllers are run. Note again that some of the controllers are both.
+
+The admission controllers intercept requests based on the values given in the configuration. In this config, we have to provide the details on 
+* What resources should it be looking for ? (Pods, Service)
+* What api version and group does it belong to ? *Example : ApiVersion = (v1, v1beta) Group version = (admissionregistration.k8s.io)*
+* What kind of operation should it intercept ? (Create, Update, Delete)
```

Comment:
Why intercept a delete? Operators commonly use finalizers to handle those scenarios.

Reply:
It is an option that is available to use if we want :) Another briefing on the fields in validatingwebhookconfiguration. Let me know if you want that removed.